StxInsider/app/src/main/kotlin/com/stxnext/stxinsider/DetailsActivity.kt

```diff
@@ -65,19 +75,36 @@ class DetailsActivity<T> : AppCompatActivity() {
             replaceImage(replaceImagePath)
     }
 
+    private fun initializeToolbar(item: DetailsItem<T>) {
+        setSupportActionBar(mToolbar)
+
+        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
+        supportActionBar!!.title = item.title.toString()
+
+        var myBitmap :Bitmap?;
+        var replaceImagePath : String? = mItem?.replacingImagePath
+        if (replaceImagePath != null) {
+            val file = this.assets.open(replaceImagePath)
+            myBitmap = BitmapFactory.decodeStream(file)
+        } else
+            myBitmap = BitmapFactory.decodeResource(resources, R.drawable.event_background);
+
+        if (myBitmap != null && !myBitmap.isRecycled) {
+            Palette.from(myBitmap).generate(Palette.PaletteAsyncListener({ palette: Palette ->
+                        val rgbValue = palette.lightMutedSwatch?.titleTextColor!!
```

Comment:
This listener returns a palette with list of best swatches to use. So it will be better to sort results by population and get more correct choice for text color.